/*
 *  Copyright (c) 2016, https://github.com/wubenqi
 *  All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package zproto;

option java_package = "com.zchat.engine.zproto";
option java_outer_classname = "ZProto";
option optimize_for = CODE_SIZE;

// TODO(@benqi): 服务端MethodID通过crc32自动生成
enum MethodID {
  METHOD_ID_UNKNOWN = 0;

  VOID_RSP = 0x01;

  USER_TOKEN_AUTH_REQ = 0xfe01;
  USER_TOKEN_AUTH_RSP = 0xfe02;

  // auths
  SERVER_AUTH_REQ = 0xff01;
  SERVER_AUTH_RSP = 0xff02;
  USER_NAME_PASSWORD_AUTH_REQ = 0xff03;
  USER_NAME_PASSWORD_AUTH_RSP = 0xff04;

  // status
  CLIENT_ONLINE_REQ = 0xff05;
  CLIENT_ONLINE_RSP = 0xff06;
  CLIENT_OFFLINE_REQ = 0xff07;
  CLIENT_OFFLINE_RSP = 0xff08;
  QUERY_ONINE_USER_REQ = 0xff19;
  QUERY_ONINE_USER_RSP = 0xff1a;

  // messages
  SEND_MESSAGE_REQ = 0xff09;
  SEQ_DATE_RSP = 0xff0a;
  MESSAGE_NOTIFY = 0xff0b;
  MESSAGE_DATA_NOTIFY = 0xff0c;
  MESSAGE_SYNC_REQ = 0xff0d;
  MESSAGE_SYNC_RSP = 0xff0e;
  MESSAGE_READ_REQ = 0xff0f;
  LOAD_HISTORY_MESSAGE_REQ = 0xff10;
  LOAD_HISTORY_MESSAGE_RSP = 0xff11;
  LOAD_DIALOGS_REQ = 0xff12;
  LOAD_DIALOGS_RSP = 0xff13;
  
  FORAWRD_MESSAGE_REQ = 0xff14;
}
